src/api/friend/friendGithubGET.ts

```diff
@@ -0,0 +1,47 @@
+import { Request, Response } from "express";
+import { success, fail } from "../../library/response";
+import { sc, rm } from "../../constants";
+import { githubDTO, friendDTO } from "../../DTO";
+import { friendService, githubService, userService } from "../../service";
+
+/**
+ * @route GET /friend/github/:username
+ * @desc 추가할 친구 깃허브 조회
+ */
+module.exports = async (req: Request, res: Response) => {
+  const userId: number = req.body.user.id;
+  const reqData: githubDTO.getReqDTO = req.params;
+  const { username } = reqData;
+
+  if (!username) return res.status(sc.BAD_REQUEST).send(fail(sc.BAD_REQUEST, rm.NULL_VALUE));
+  
+  try {
+    const { profileImage } = await githubService.getProfileImage(username);
+
+    if (!profileImage) return res.status(sc.NOT_FOUND).send(fail(sc.NOT_FOUND, rm.NO_GITHUB));
+
+    const users = await userService.getUserByUsername(username);
+
+    if (users.length === 0) return res.status(sc.NOT_FOUND).send(fail(sc.NOT_FOUND, rm.NO_USER));
+
+    const resData = await Promise.all(
+      users.map(async o => {
+        const isFriend = await friendService.getFriend(userId, o.id) ? true : false;
+        const profile: friendDTO.IProfile = {
+          id: o.id,
+          username: o.username,
+          nickname: o.nickname,
```

Comment:
매핑할 때 o 대신 user는 어떠신지요 ㅋㅋ 좀 더 알아보기 쉽게..?
```suggestion
      users.map(async user => {
        const isFriend = await friendService.getFriend(userId, user.id) ? true : false;
        const profile: friendDTO.IProfile = {
          id: user.id,
          username: user.username,
          nickname: user.nickname,
```

Reply:
조아용~